securebanking-openbanking-uk-rcs-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/api/ConsentDetailsApiController.java

```diff
@@ -94,21 +104,33 @@ public ResponseEntity<ConsentDetails> getConsentDetails(String consentRequestJws
             log.debug("Intent Id from the requested claims '{}'", intentId);
 
             ConsentRequest consentRequest = buildConsentRequest(signedJWT);
+            log.debug("Intent type: '{}' with ID '{}'", IntentType.identify(intentId), intentId);
 
-            if (IntentType.identify(intentId) != null) {
-                log.debug("Intent type: '{}' with ID '{}'", IntentType.identify(intentId), intentId);
+            IntentType intentType = IntentType.identify(intentId);
+            if (intentType != null) {
+                log.debug("Intent type: '{}' with ID '{}'", intentType, intentId);
                 log.debug("Retrieve consent details:\n- Type '{}'\n-Id '{}'\n",
-                        IntentType.identify(consentRequest.getIntentId()).name(), consentRequest.getIntentId());
+                        intentType.name(), consentRequest.getIntentId());
                 JsonObject consent = consentServiceClient.getConsent(consentRequest);
 
                 log.debug("Retrieve to api client details for client Id '{}'", consentRequest.getClientId());
                 ApiClient apiClient = apiClientService.getApiClient(consentRequest.getClientId());
                 log.debug("ApiClient controller: " + apiClient);
                 log.debug("consent json controller: " + consent);
-                // build the consent details object for the response
-                ConsentDetails consentDetails = ConsentDetailsBuilderFactory.build(consent, consentRequest, apiClient);
-
-                return ResponseEntity.ok(consentDetails);
+                // consent details object by intent type
+                ConsentDetails consentDetailsToDisplay = consentDetailsFactory.getConsentDetails(intentType);
```

Comment:
is this threadsafe?

Reply:
Good point!, isn't thread safe, only the enumMap is thread safe for the scope, not the instances returned by the factory.

I did some changes to make it thread safe, now the factory returns a consentDetails instance by intent type to ensure thread safe.

Reply:
Deleted, now the factory provider returns a factory instance of type ConsentDetailsFactory to decode the json depending of the IntentType.